// scube-cms/config/database-config.js
// Expose the database configuration to our app directly via module.exports

module.exports = {

  'openshift': {
    prd: {
      host     : process.env.OPENSHIFT_MYSQL_DB_HOST,
      username : 'scube_prd',
      password : 'thirst_to_app_store_nov@2015',
      port     : process.env.OPENSHIFT_MYSQL_DB_PORT,
      database : 'scube'
    },

    dev: {
      host     : '127.0.0.1',
      username : 'scube_admin',
      password : 'insvikrant+*-/2015',
      port     : 3306,
      database : 'scube_dev'
    }
  },

  'openshift_mongo': {
    prd: {
      host     : process.env.OPENSHIFT_MONGODB_DB_HOST,
      username : 'scube_admin',
      password : 'insvikrant+*-/2015',
      port     : process.env.OPENSHIFT_MONGODB_DB_PORT,
      database : 'scube_mongodb',
      appname  : 'parkking'
    },

    dev: {
      host     : '127.0.0.1',
      username : 'scube_admin',
      password : 'insvikrant+*-/2015',
      port     : 27017,
      database : 'scube_mongodb_dev',
      appname  : 'parkking'
    }
  }
};
