// scube-cms/defines/dir.js
/*********************Global File for all Consumer APP Scube directory paths*************************/
///////////////////////////////////////////////////////////////////////////////////////////

/*******************************Scube Library Paths BEGIN*************************************/
//Logger path
exports.LOGGER                          =  'scube-cms/libs/logger/logger';
//Database Paths
exports.DB_SP                           =  'scube-cms/libs/database/store-procedure';
exports.DB_CONNECT                      =  'scube-cms/libs/database/mysql-db';
//Util Paths
exports.HASHER                          =  'scube-cms/libs/utils/hasher';
exports.MAILER                          =  'scube-cms/libs/utils/mailer';
exports.JSON_UTIL                       =  'scube-cms/libs/utils/json';
exports.DOMAIN_HANDLER                  =  'scube-cms/libs/utils/domain-handler';
exports.DATA_PARSER                     =  'scube-cms/libs/utils/data-parser';
exports.JSON_BUILDER                    =  'scube-cms/libs/utils/json-builder';
exports.IMAGE_LIB                       =  'scube-cms/libs/utils/image-lib';
//GCM Paths
exports.GCM_POST                        =  'scube-cms/libs/gcm/gcm-post';
/*******************************Scube Library Paths END***************************************/

/*******************************Scube Config Paths START**************************************/
exports.DB_CONFIG                       =  'scube-cms/config/database-config';
exports.SOCIAL_CONFIG                   =  'scube-cms/config/social-config';
exports.GCM_CONFIG                      =  'scube-cms/config/gcm-config';
/*******************************Scube Library Paths END***************************************/

/*******************************Scube Services Paths START************************************/
exports.SERVICES_LOCATION               =  'scube-cms/services/locations-service';
exports.SERVICES_DEVICE                 =  'scube-cms/services/device-service';
exports.SERVICES_DEALS                  =  'scube-cms/services/deals-service';
/*******************************Scube Routes Paths START************************************/
//R implies its routes

//Used in APP.js file
exports.R_CMS_COMBO_HELPERS             = './routes/cms-app/combo-helpers/'
exports.R_CMS_MALL                      = './routes/cms-app/malls/'
exports.R_CMS_BRAND_PROFILE             = './routes/cms-app/brand-profile/'
exports.R_CMS_BRANDS                    = './routes/cms-app/brands/'
exports.R_CMS_DEVICE                    = './routes/cms-app/device/'
exports.R_CMS_DEVICE_LOGIN              = './routes/cms-app/device-login/'
exports.R_CMS_DEVICE_OS                 = './routes/cms-app/device-os/'
exports.R_CMS_DEVICE_TYPE               = './routes/cms-app/device-type/'
exports.R_CMS_DEVICE_VARIANT            = './routes/cms-app/device-variant/'
exports.R_CMS_DEVICE_LOCATION           = './routes/cms-app/location/'
exports.R_CMS_POP_INDEX                 = './routes/cms-app/popularity-index/'
exports.R_CMS_SCUBE_DEALS               = './routes/cms-app/scube-deals/'
exports.R_CMS_SCUBIT                    = './routes/cms-app/scubit/'
exports.R_CMS_SHOP_PROFILE              = './routes/cms-app/shop-profile/'
exports.R_CMS_SHOPPING_CATEGORY         = './routes/cms-app/shopping-category/'
exports.R_CMS_SHOPPING_OFFER            = './routes/cms-app/shopping-offer/'
exports.R_CMS_SHOP                      = './routes/cms-app/shop/'
exports.R_CMS_USER_BRAND_PREF           = './routes/cms-app/user-brand-pref'
exports.R_CMS_USER_CAT_PREF             = './routes/cms-app/user-category-pref'
exports.R_CMS_USER_LOGIN                = './routes/cms-app/user-login'
exports.R_CMS_USER_PROFILE              = './routes/cms-app/user-profile'
exports.R_CMS_HOMEPAGE                  = './routes/cms-app/homepage'

// Image CMS Routes
exports.R_CMS_IMAGES_MALLS              = './routes/cms-images/malls/'
exports.R_CMS_IMAGES_SHOPS              = './routes/cms-images/shops/'
exports.R_CMS_IMAGES_BRANDS             = './routes/cms-images/brands/'
exports.R_CMS_IMAGES_SHOP_PROFILE       = './routes/cms-images/shop-profile/'
exports.R_CMS_IMAGES_HOMEPAGE           = './routes/cms-images/homepage/'
/*******************************Scube Routes Paths END**************************************/
